README.md

```diff
@@ -1105,6 +1110,59 @@ However, the following rules of thumb are useful to keep in mind:
   Alternatively, the entire OrientDB class and all edges that may point to it may be omitted
   entirely from the GraphQL schema.
 
+## Query cost estimation
+
+Due to GraphQL's easy-to-use nature, queries that use a lot of memory and/or have a long execution
+time can happen often. For better user-friendliness, it's worthwhile to estimate a query's cost (in
+terms of time taken and memory used) using statistics without executing the query on the database.
+Since the time and memory required for query execution varies depending on the underlying database,
+as a rough proxy, the cost estimator estimates the number of result rows for the given query i.e.
+the query's cardinality.
+
+These estimates can be used to improve user-friendliness for query execution.
+For example, the estimates can be used to warn users about a query that asks for too many results.
+
+### Usage of query cost estimation
+
+To get estimates for the number of result rows for a given query and statistics object, use the
+`estimate_query_result_cardinality()` function in
+`graphql_compiler/cost_estimation/cardinality_estimator.py`.
+
+### Statistics
+
+Before estimates can be generated, you need to provide statistics about the graph data. The
+applicable classes are in `graphql_compiler/cost_estimation/statistics.py`. Currently, two classes
+are implemented:
+- `Statistics`, an abstract base class that describes the interface the cost estimator expects for
+  statistics. This allows users to create their own implementations for the statistic class.
+- `LocalStatistics`, an implementation of `Statistics`, where the statistics are provided during
+  initialization, and can't be modified thereafter.
+
+Currently, three statistics are supported. `get_class_count()` must be provided for cardinality
+estimation. The other two are optional, as they can be roughly approximated.
+
+### Required statistics
+
+- `get_class_count(class)`: 
```

Comment:
This can be best shown through a code example than described here.

Reply:
I'll also include an example. The current documentation on statistics can't hurt, can it?

Reply:
It can hurt. The README.md should be as concise as possible. Furthermore, this section should fit the style of the rest of the README.md. I don't think that class documentation is mentioned anywhere else, (and for good reason).

Reply:
As we move to read the docs formatting, is it possible that all of this stuff will get moved to there (where conciseness matters less) and the README ends up much shorter than it is now? In that case maybe we don't need to worry about being concise as much.